store/mongo/migration_2_0_0.go

```diff
@@ -0,0 +1,179 @@
+	mopts "go.mongodb.org/mongo-driver/mongo/options"
+)
+
+const (
+	findBatchSize    = 100
+	IndexUserIdEmail = "UserIdEmail"
+)
+
+type migration_2_0_0 struct {
+	ds     *DataStoreMongo
+	dbName string
+	ctx    context.Context
+}
+
+func (m *migration_2_0_0) Up(from migrate.Version) error {
+	logger := log.FromContext(m.ctx)
+	ctx := context.Background()
+
+	collectionsIndexes := map[string]struct {
+		Indexes []mongo.IndexModel
+	}{
+		DbUsersColl: {
+			Indexes: []mongo.IndexModel{
+				{
+					Keys: bson.D{{Key: DbUserEmail, Value: 1}},
+					Options: mopts.Index().
+						SetUnique(true).
+						SetName(DbUniqueEmailIndexName),
+				},
+			},
+		},
+		DbTokensColl: {
+			Indexes: []mongo.IndexModel{
+				{
+					Keys: bson.D{
+						{Key: DbTokenExpiresAt, Value: 1},
+					},
+					Options: mopts.Index().
+						SetExpireAfterSeconds(0).
+						SetName(IndexUserIdEmail),
+				},
+				{
+					Keys: bson.D{
+						{Key: mstore.FieldTenantID, Value: 1},
+						{Key: DbTokenSubject, Value: 1},
+						{Key: DbTokenName, Value: 1},
+					},
+					Options: mopts.Index().
+						SetUnique(true).
+						SetPartialFilterExpression(
+							bson.M{
+								DbTokenName: bson.M{"$exists": true},
+							}).
+						SetName(DbTenantUniqueTokenNameIndexName),
+				},
+				{
+					Keys: bson.D{
+						{Key: mstore.FieldTenantID, Value: 1},
+						{Key: DbTokenSubject, Value: 1},
+					},
+					Options: mopts.Index().
+						SetName(DbTenantTokenSubjectIndexName),
+				},
+			},
+		},
+		DbSettingsColl: {
+			Indexes: []mongo.IndexModel{},
+		},
+	}
+
+	// for each collection in main useradm database
+	if m.dbName == DbName {
+		for collection, indexModel := range collectionsIndexes {
+			coll := m.ds.client.Database(m.dbName).Collection(collection)
+			// drop all the existing indexes, ignoring the errors
+			_, _ = coll.Indexes().DropAll(ctx)
+
+			// create the new indexes
+			if len(indexModel.Indexes) != 0 {
+				_, err := coll.Indexes().CreateMany(ctx, indexModel.Indexes)
+				if err != nil {
+					return err
+				}
+			}
+		}
+	}
+
+	tenantID := mstore.TenantFromDbName(m.dbName, DbName)
+	ctx = identity.WithContext(ctx, &identity.Identity{
+		Tenant: tenantID,
+	})
+
+	// for each collection
+	for collection := range collectionsIndexes {
+		coll := m.ds.client.Database(m.dbName).Collection(collection)
+		collOut := m.ds.client.Database(DbName).Collection(collection)
+		writes := make([]mongo.WriteModel, 0, findBatchSize)
+
+		if m.dbName == DbName {
+			// if any documents already exist in "useradm" ds,
+			// add empty "tenant_id": "" key-value pair
+			tenantIdFilter := bson.D{
+				{Key: mstore.FieldTenantID, Value: bson.D{{Key: "$exists", Value: false}}}}
+			update := bson.M{"$set": bson.M{mstore.FieldTenantID: ""}}
+			result, err := collOut.UpdateMany(ctx, tenantIdFilter, update)
+			logger.Infof("Modified documents in main useradm database count: %d",
+				result.ModifiedCount)
+			if err != nil {
+				return err
+			}
```

Comment:
this will add tenant_id to all the settings items in the useradm db on hM and other collections too, maybe we should verify that it is fine? (I have no idea how to verify since Manuel is not present)

Reply:
I think this should be transparent for the API users (like Manuel ;)), right?

Reply:
I hope so.